lib/_http_client.js

```diff
@@ -260,11 +271,17 @@ function ClientRequest(input, options, cb) {
   }
 
   const oncreate = (err, socket) => {
-    if (called)
+    const state = this[kWritableState];
+    if (!state.socketPending)
       return;
-    called = true;
+    state.socketPending = false;
     if (err) {
-      process.nextTick(() => this.emit('error', err));
+      const { destroyCallback: cb } = state;
+      if (cb) {
+        cb(err);
+      } else {
+        this.destroy(err);
+      }
```

Comment:
Shouldn’t we call destroy(err, cb) instead?

Reply:
Why do you think that? If we have `cb` then we're already in the middle of a `destroy`.

Reply:
See `_destroy`